### PR TITLE
MAISTRA-1093 Ensure all tests use a client with all resource kinds us…

### PR DESCRIPTION
**Pull request by luksa**
…ed by the operator

### CODE REVIEW

File: pkg/controller/common/test/controller-test-utils.go [hunk 1]
```diff
@@ -5,18 +5,47 @@ import (
...
+		Version: "v1",
+		Kind:    "NetworkAttachmentDefinition",
+	}, &unstructured.Unstructured{})
+	s.AddKnownTypeWithName(schema.GroupVersionKind{
+		Group:   "k8s.cni.cncf.io",
+		Version: "v1",
+		Kind:    "NetworkAttachmentDefinitionList",
+	}, &unstructured.UnstructuredList{})
+
+	tracker := clienttesting.NewObjectTracker(s, scheme.Codecs.UniversalDecoder())
```
Comment (rcernich) [inline pkg/controller/common/test/controller-test-utils.go]: I think this should be initialized with a serializer created from the scheme, as opposed to the generic runtime scheme.

Comment (luksa) [reply pkg/controller/common/test/controller-test-utils.go]: I've changed this now. Can you see if it's what you meant?

**Review by rcernich — APPROVED** [review]

File: pkg/controller/common/test/controller-test-utils.go [hunk 2]
```diff
@@ -5,18 +5,49 @@ import (
...
+	"github.com/maistra/istio-operator/pkg/apis"
 	"github.com/maistra/istio-operator/pkg/controller/common/test/assert"
 )
 
+func CreateClient(clientObjects ...runtime.Object) (client.Client, *EnhancedTracker) {
+	s := scheme.Scheme                          // scheme must be initialized before creating the client below
+	if err := apis.AddToScheme(s); err != nil { // scheme must be initialized before creating the client below
+		panic(fmt.Sprintf("Could not add to scheme: %v", err))
+	}
+	if err := rbac.AddToScheme(s); err != nil { // scheme must be initialized before creating the client below
```
Comment (dgn) [inline pkg/controller/common/test/controller-test-utils.go]: nit: let's remove these comments now

**Review by dgn — APPROVED** [review]

File: pkg/controller/common/test/controller-test-utils.go (hunk 2 above)
Comment (luksa) [reply pkg/controller/common/test/controller-test-utils.go]: Done. Thanks

**Pull request merged**